discourse_rock/patches/rake.patch

```
@@ -0,0 +1,87 @@
+--- /dev/null
++++ b/lib/tasks/discourse-charm.rake
+@@ -0,0 +1,88 @@
++# frozen_string_literal: true
++
++desc "Promote a user to admin"
++task "admin:promote", [:email] => [:environment] do |_, args|
++email = args[:email]
++  if !email || email !~ /@/
++    puts "ERROR: Expecting rake admin:promote[some@email.com]"
++    exit 1
++  end
++  if User.find_by_email(email).nil?
++    puts "ERROR: User with email #{email} not found"
++    exit 1
++  end
++ admin = User.find_by_email(email)
++ admin.grant_admin!
++ admin.change_trust_level!(1) if admin.trust_level < 1
++ admin.email_tokens.update_all confirmed: true
++ admin.activate
++ say("\nYour account now has Admin privileges!")
++end
++
++desc "Create the user for given credentials"
++task "users:create" => :environment do
++  require "highline/import"
++
++  begin
++    email = ask("Email:  ")
++    existing_user = User.find_by_email(email)
++
++    # check if user account already exists
++    if existing_user
++      # user already exists, ask for password reset
++      new_user = existing_user
++      reset_password =
++        ask(
++          "User with this email already exists! Do you want to reset the password for this email? (Y/n)  ",
```

I would return an error and add a flag to reset password if exists